pkg/scheduler/scheduling_partition_test.go

```diff
@@ -43,6 +43,12 @@ func newTestPartition() (*partitionSchedulingContext, error) {
 	return newPartitionSchedulingContext(info, rootSched), nil
 }
 
+func newTestPartitionProtected(t *testing.T) *partitionSchedulingContext {
+	partition, err := newTestPartition()
+	assert.NilError(t, err, "test partition create failed with error")
+	return partition
+}
+
```

Comment:
Same as for the info objects, don't hide the error in a function.
This reversal will also require variable declaration to be reversed to prevent shadowing